### PR TITLE
added solution - hash

### PR DESCRIPTION
**Pull request by glones**

### CODE REVIEW

File: src/main/resources/init_db.sql [hunk 1]
```diff
@@ -122,3 +122,6 @@ VALUES (1, 1);
 
 INSERT INTO `internet_shop`.`users_roles` (user_id, role_id)
 VALUES (2, 2);
+
+ALTER TABLE `internet_shop`.`users`
+ADD COLUMN `salt` VARBINARY(16) NOT NULL AFTER `password`;
```
Comment (voloshynamaryna11) [inline src/main/resources/init_db.sql:127]: лучше просто добавь в таблицу новый столбец

**Review by voloshynamaryna11 — COMMENTED** [review]
good

File: src/main/java/com/internet/shop/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format(FORMAT_PATTERN, b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Problem with password hashing");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/util/HashUtil.java:28]:
```suggestion
            throw new RuntimeException("Problem with password hashing", e);
```

File: src/main/resources/init_db.sql (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/resources/init_db.sql:127]: I agree, let's better modify CREATE statement

also, I see that you are inserting users here
let's do this in injectDataController (you should insert salt to DB as well)

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/resources/init_db.sql (hunk 1 above)
Comment (glones) [reply src/main/resources/init_db.sql:127]: done

File: src/main/java/com/internet/shop/util/HashUtil.java (hunk 2 above)
Comment (glones) [reply src/main/java/com/internet/shop/util/HashUtil.java:28]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**